virtual-desktop/src/app/shared/language-locale.service.ts

```diff
@@ -0,0 +1,79 @@
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+import { Injectable /*, Inject */ } from '@angular/core';
+import { Http } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/catch';
+import { fromPromise } from 'rxjs/observable/fromPromise';
+
+import { Globalization } from './globalization';
+
+@Injectable()
+export class LanguageLocaleService {
+
+  private readonly plugin: ZLUX.Plugin = ZoweZLUX.pluginManager.getDesktopPlugin();
+  private readonly logger = ZoweZLUX.logger.makeComponentLogger(this.plugin.getIdentifier());
```

Comment:
Future: objects in desktop shouldnt be making a new component logger, but rather passing around 1 for all and making subloggers off it.